src/Microsoft.ML.FastTree/FastTreeRegression.cs

```diff
@@ -124,7 +124,7 @@ protected override ObjectiveFunctionBase ConstructObjFunc(IChannel ch)
             return new ObjectiveImpl(TrainSet, Args);
         }
 
-        protected override OptimizationAlgorithm ConstructOptimizationAlgorithm(IChannel ch)
+        internal override OptimizationAlgorithm ConstructOptimizationAlgorithm(IChannel ch)
```

Comment:
>internal [](start = 8, length = 8)

Another candidate for `private protected`.

Reply:
All `ConstructOptimizationAlgorithm` are `private protected` now because we have
```csharp
    public abstract class FastTreeTrainerBase<TArgs, TTransformer, TModel>
    ....
        private protected abstract OptimizationAlgorithm ConstructOptimizationAlgorithm(IChannel ch);
    ....
```

---
In reply to: [251652163](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251652163) [](ancestors = 251652163)